finquant/portfolio.py

```diff
@@ -1064,8 +1066,8 @@ def _build_portfolio_from_df(
         name = pf_allocation.iloc[i].Name
         # extract data column of said stock
         stock_data = data.loc[:, [name]].copy(deep=True).squeeze()
-        # create Stock instance and add it to portfolio
-        pf.add_stock(Stock(pf_allocation.iloc[i], data=stock_data))
+        # create Stock instance and add it to portfolio, defer updating until all stocks are added
```

Comment:
```suggestion
        # create Stock instance and add it to portfolio,
        # and defer updating portfolio attributes until all stocks are added
```
just a minor change in the comment here :)